core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -240,15 +245,25 @@ public void finished(Description description) {
             ambassadorContainers.forEach((String address, AmbassadorContainer container) -> container.stop());
 
             // Kill the services using docker-compose
-            getDockerCompose("down -v")
-                    .start();
+            try {
+                getDockerCompose("down -v").invoke();
 
-            // remove the networks before removing the containers
-            ResourceReaper.instance().removeNetworks(identifier);
+                // If we reach here then docker-compose down has cleared networks and containers;
+                //  we can unregister from ResourceReaper
+                spawnedContainerIds.forEach(id -> ResourceReaper.instance().unregisterContainer(id));
```

Comment:
FYI `spawnedContainerIds.forEach(ResourceReaper.instance()::unregisterContainer);` is also valid here :)